src/protocols/valve/protocol.rs

```diff
@@ -304,7 +303,7 @@ impl ValveProtocol {
         let environment_type = Environment::from_gldsrc(buffer.read()?)?;
         let has_password = buffer.read::<u8>()? == 1;
         let vac_secured = buffer.read::<u8>()? == 1;
-        let the_ship = match *engine == SteamApp::THESHIP.as_engine() {
+        let the_ship = match *engine == Engine::new(2400) {
```

Comment:
These direct comparisons with hard-coded app IDs seem like they could use an abstraction.

maybe something like:
```rs
impl Engine {
  fn is_the_ship(&self) -> bool;
  fn has_no_size_field(&self) -> bool;
  fn should_remove_test_rule(&self) -> bool;
}
```

Reply:
Interesting idea, will think about some stuff like this.
Just to be picky: `is_the_ship` wouldn't be really good, we currently reference in the valve protocol the following games: CSS (230), The Ship (2400) and ROR2 (632360), I do have an idea tho, will make a separate PR.